src/main/scala/org/graphframes/lib/ConnectedComponents.scala

```diff
@@ -336,8 +338,11 @@ object ConnectedComponents extends Logging {
 
         // remove previous checkpoint
         if (iteration > checkpointInterval) {
-          FileSystem.get(sc.hadoopConfiguration)
-            .delete(new Path(s"${checkpointDir.get}/${iteration - checkpointInterval}"), true)
+          val path = new Path(s"${checkpointDir.get}/${iteration - checkpointInterval}")
+          if (fileSystem == null) {
```

Comment:
how expensive is it to obtain a file system object? I suspect it is negligible compared to the cost of other Spark operations. Let's us not store it unless it is necessary.